alfred-telemetry-solr/alfred-telemetry-solr-common/src/main/java/eu/xenit/alfred/telemetry/solr/monitoring/binder/SolrTrackerMetrics.java

```diff
@@ -27,95 +27,113 @@ public SolrTrackerMetrics(AlfrescoCoreAdminHandler coreAdminHandler) {
-                    .getTrackerState();
+
 
             // technically these metrics are not per core, but in order to filter in grafana the core is added as a tag
             Tags tags = Tags.of("core", coreName, "state", "Remaining");
-            Gauge.builder("alfresco.transactions.nodes", metadataTrackerState,
-                    x -> getTransactionsRemaining(metadataTrackerState))
+            Gauge.builder("alfresco.transactions.nodes", coreAdminHandler,
+                            x -> getTransactionsRemaining(x, coreName))
                     .tags(tags)
                     .register(registry);
 
             tags = Tags.of("core", coreName);
-            Gauge.builder("alfresco.transactions.nodes.lag", metadataTrackerState, x -> getTxLag(metadataTrackerState))
+            Gauge.builder("alfresco.transactions.nodes.lag", coreAdminHandler, x -> getTxLag(x, coreName))
                     .tags(tags)
                     .register(registry);
 
             tags = Tags.of("core", coreName);
-            Gauge.builder("alfresco.transactions.nodes.lastIndexCommitTime", metadataTrackerState,
-                    x -> getLastIndexTxCommitTime(metadataTrackerState))
+            Gauge.builder("alfresco.transactions.nodes.lastIndexCommitTime", coreAdminHandler,
+                            x -> getLastIndexTxCommitTime(x, coreName))
                     .tags(tags)
                     .register(registry);
 
             tags = Tags.of("core", coreName, "state", "Remaining");
-            Gauge.builder("alfresco.transactions.acls", aclsTrackerState, x -> getChangeSetsRemaining(aclsTrackerState))
+            Gauge.builder("alfresco.transactions.acls", coreAdminHandler, x -> getChangeSetsRemaining(x, coreName))
                     .tags(tags).register(registry);
 
             tags = Tags.of("core", coreName);
-            Gauge.builder("alfresco.transactions.acls.lag", aclsTrackerState, x -> getChangeSetsLag(aclsTrackerState))
+            Gauge.builder("alfresco.transactions.acls.lag", coreAdminHandler, x -> getChangeSetsLag(x, coreName))
                     .tags(tags)
                     .register(registry);
 
             tags = Tags.of("core", coreName);
-            Gauge.builder("alfresco.transactions.acls.lastIndexCommitTime", aclsTrackerState,
-                    x -> getLastIndexChangeSetCommitTime(aclsTrackerState))
+            Gauge.builder("alfresco.transactions.acls.lastIndexCommitTime", coreAdminHandler,
+                            x -> getLastIndexChangeSetCommitTime(x, coreName))
                     .tags(tags)
                     .register(registry);
         }
     }
 
-    private long getLastIndexChangeSetCommitTime(TrackerState aclsTrackerState) {
-        return aclsTrackerState.getLastIndexedChangeSetCommitTime();
+    private static TrackerRegistry getTrackerRegistry(AlfrescoCoreAdminHandler coreAdminHandler) {
```

Comment:
what do you save by adding this method instead of using the implementation directly?